### PR TITLE
feat!: Change Hook Config field from map to *HookConfig

### LINKED ISSUE #3072: Hook should contain Config of type HookConfig
The `CreateHook` method
https://github.com/google/go-github/blob/6ca4d142bef57e88f87290837796a54a0296fcc4/github/repos_hooks.go#L85

currently wants a `Hook` parameter that has `Config` of type `map[string]interface{}`
https://github.com/google/go-github/blob/6ca4d142bef57e88f87290837796a54a0296fcc4/github/repos_hooks.go#L54

This forces the user to manually define each config properties.

My proposal is to change the `Config` to `HookConfig`, that for some reasons I'm not aware of is defined here:

https://github.com/google/go-github/blob/6ca4d142bef57e88f87290837796a54a0296fcc4/github/orgs_audit_log.go#L24-L31

This could be a breaking change for people currently using the `CreateHook` function (and every other functions taking `Hook` as input) so I am asking first how you want to proceed about it.

### PR DESCRIPTION
**Pull request by himazawa**
Fixes: #3072.

BREAKING CHANGE: Changes `Hook.Config` from `map[string]interface{}` to `*HookConfig`.

### CODE REVIEW

**Comment by himazawa** [comment]
I would like to sign the CLA but there is no way to add the `noreply.github.com` email to the CLA account.

File: github/hooks_config.go
Comment (gmlewis) [inline github/hooks_config.go:3]: Instead of creating a file with only one struct in it, let's please move this to `repos_hooks_configuration.go` where it is most heavily used.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @himazawa.

**Comment by gmlewis** [comment]
> I would like to sign the CLA but there is no way to add the `noreply.github.com` email to the CLA account.

Did you read the instructions for email addresses here: https://github.com/google/go-github/pull/3073/checks?check_run_id=21540833172 ?

**Comment by gmlewis** [comment]
Well, looks like you got it resolved whatever the problem was.

**Comment by gmlewis** [comment]
Please run step 4 of CONTRIBUTING.md and then push (not force-push) the changes to this PR.

**Comment by himazawa** [comment]
Done

**Review by gmlewis — APPROVED** [review]
Thank you, @himazawa !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
